src/python/dxpy/dx_extract_utils/ExpressionInputsValidator.py

```diff
@@ -0,0 +1,179 @@
+import dxpy
+
+from ..exceptions import (
+    err_exit,
+    PermissionDenied,
+    InvalidInput,
+    InvalidState,
+    ResourceNotFound,
+    default_expected_exceptions,
+)
+from ..utils.resolver import resolve_existing_path, is_hashid, ResolutionError
+
+import json
+import os
+
+
+class ExpressionInputsValidator:
+    """InputsValidator class for extract_assay expresion. Checks for invalid input combinations"""
+
+    def __init__(self, args):
+        self.error_handler = None
+
+        self.list_assays = args.list_assays
+        self.retrieve_expression_flag = args.retrieve_expression
+        self.assay_name = args.assay_name
+        self.input_json = args.input_json
+        self.input_json_file = args.input_json_file
+        self.json_help = args.json_help
+        self.sql = args.sql
+        self.additional_fields = args.additional_fields
+        self.additional_fields_help = args.additional_fields_help
+        self.expression_matrix = args.expression_matrix
+        self.delim = args.delim
+        self.output = args.output
+
+        self.validations_to_run()
+
+    def validations_to_run(self):
+        self.end_arguments_check("list_assays")
+        self.end_arguments_check("additional_fields_help")
+
+        self.forbiden_combination_validation(
+            invalid_combination=self.json_help
+            and any(
+                [
+                    self.assay_name,
+                    self.sql,
+                    self.additional_fields,
+                    self.expression_matrix,
+                    self.output,
+                ]
+            ),
+            error_message='"--json-help" cannot be passed with any of "--assay-name", "--sql", "--additional-fields", "--expression-matrix", or "--output"',
+        )
+        self.forbiden_combination_validation(
+            invalid_combination=self.retrieve_expression_flag
+            and not any([self.input_json, self.json_help]),
+            error_message='The flag, "--retrieve_expression" must be followed by a json input or json help.',
+        )
+        self.forbiden_combination_validation(
+            invalid_combination=(self.input_json == "{}"),
+            error_message='JSON for "--retrieve-expression" does not contain valid filter information.',
+        )
+
+    def end_arguments_check(self, end_arg):
+        args_specific_list = [
+            getattr(self, arg)
+            for arg in dir(self)
+            if not callable(getattr(self, arg))
+            and not arg.startswith("__")
+            and arg != end_arg
+        ]
+
+        invalid_combination = getattr(self, end_arg) and any(args_specific_list)
+
+        if invalid_combination:
+            self.error_handler = (
+                '"--{}" cannot be presented with other options.'.format(
+                    end_arg.replace("_", "-")
+                )
+            )
+
+    def forbiden_combination_validation(self, invalid_combination, error_message):
```

Comment:
I would personally get rid of this method entirely and simply let all messages be handled by the `error_handler`.